pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationDataSource.java

```diff
@@ -82,39 +83,77 @@ default String getHttpHeader(String name) {
 
     /**
      * Check if data from Pulsar protocol are available.
-     * 
+     *
      * @return true if this authentication data contain data from Pulsar protocol
      */
     default boolean hasDataFromCommand() {
         return false;
     }
 
     /**
-     * 
+     *
      * @return authentication data which is stored in a command
      */
     default String getCommandData() {
         return null;
     }
 
+
+    /*
```

Comment:
I think the newly introduced methods can just be part of SaslAuthenticationDataSource. Since they are only used by SaslAuthenticationProvider and SaslAuthenticationProvider will check if a data source is a sasl data source, it can cast the data source and access those methods. 

We don't need to put the methods here.